assets/js/src/core/components/horizontal-scroll/horizontal-scroll.tsx

```diff
@@ -0,0 +1,154 @@
+/**
+* Pimcore
+*
+* This source file is available under two different licenses:
+* - Pimcore Open Core License (POCL)
+* - Pimcore Commercial License (PCL)
+* Full copyright and license information is available in
+* LICENSE.md which is distributed with this source code.
+*
+*  @copyright  Copyright (c) Pimcore GmbH (http://www.pimcore.org)
+*  @license    https://github.com/pimcore/studio-ui-bundle/blob/1.x/LICENSE.md POCL and PCL
+*/
+
+import React, { useEffect, useRef, useState } from 'react'
+import { useStyles } from './horizontal-scroll.styles'
+import { IconButton } from '@Pimcore/components/icon-button/icon-button'
+import { Flex } from 'antd'
+
+export interface HorizontalScrollProps {
+  children: React.ReactNode
+  scrollWidth?: number
+}
+
+export const HorizontalScroll = ({ children, scrollWidth }: HorizontalScrollProps): React.JSX.Element => {
+  const scrollContainerRef = useRef<HTMLDivElement>(null)
+  const scrollAmount = 50
+  const scrollSpeed = 30
+  const [scrollInterval, setScrollInterval] = useState<NodeJS.Timeout | null>(null)
+  const [isAtStart, setIsAtStart] = useState(true)
+  const [isAtEnd, setIsAtEnd] = useState(false)
+  const [scrollRequired, setScrollRequired] = useState(false)
+  const [hideElement, setHideElement] = useState(false)
+  const { styles } = useStyles({ scrollWidth, hideElement })
+
+  const updateScrollState = (): void => {
+    if (scrollContainerRef.current !== null) {
+      const { scrollLeft, scrollWidth, clientWidth } = scrollContainerRef.current
+
+      setIsAtStart(scrollLeft === 0)
+      setIsAtEnd(scrollLeft + clientWidth >= scrollWidth)
+      setScrollRequired(scrollContainerRef.current.scrollWidth > scrollContainerRef.current.clientWidth)
+      setHideElement(scrollContainerRef.current.clientWidth < 50)
+    }
+  }
+
+  useEffect(() => {
+    if (scrollContainerRef.current !== null) {
+      updateScrollState()
+      scrollContainerRef.current.addEventListener('scroll', updateScrollState)
+
+      const resizeObserver = new ResizeObserver(() => {
+        updateScrollState()
+      })
+
+      resizeObserver.observe(scrollContainerRef.current)
+
+      return () => {
```

Comment:
I think we should also clear the scrollInterval here. In case there is a weired bug this interval would never be cleared.

Reply:
agree - and have just added it